MotionTrackingRajawali/app/src/main/java/com/projecttango/experiments/motiontrackingrajawali/MotionTrackingActivity.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2014 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.projecttango.experiments.motiontrackingrajawali;
+
+import com.google.atap.tangoservice.Tango;
+import com.google.atap.tangoservice.Tango.OnTangoUpdateListener;
+import com.google.atap.tangoservice.TangoConfig;
+import com.google.atap.tangoservice.TangoCoordinateFramePair;
+import com.google.atap.tangoservice.TangoErrorException;
+import com.google.atap.tangoservice.TangoEvent;
+import com.google.atap.tangoservice.TangoOutOfDateException;
+import com.google.atap.tangoservice.TangoPoseData;
+import com.google.atap.tangoservice.TangoXyzIjData;
+
+import android.app.Activity;
+import android.os.Bundle;
+import android.widget.Toast;
+
+import org.rajawali3d.surface.IRajawaliSurface;
+import org.rajawali3d.surface.RajawaliSurfaceView;
+
+import java.util.ArrayList;
+
+/**
+ * Main Activity class for the Motion Tracking Rajawali Sample. Handles the connection to the Tango
+ * service and propagation of Tango pose data to OpenGL and Layout views. OpenGL rendering logic is
+ * delegated to the {@link MotionTrackingRajawaliRenderer} class.
+ */
+public class MotionTrackingActivity extends Activity {
+
+    private Tango mTango;
+    private TangoConfig mConfig;
+    private MotionTrackingRajawaliRenderer mRenderer;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.activity_motion_tracking);
+        mTango = new Tango(this);
+        mConfig = setupTangoConfig(mTango, true);
+        mRenderer = setupGLViewAndRenderer();
+    }
+
+    private MotionTrackingRajawaliRenderer setupGLViewAndRenderer(){
+        MotionTrackingRajawaliRenderer renderer = new MotionTrackingRajawaliRenderer(this);
+        RajawaliSurfaceView glView = (RajawaliSurfaceView) findViewById(R.id.gl_surface_view);
+        glView.setEGLContextClientVersion(2);
+        glView.setRenderMode(IRajawaliSurface.RENDERMODE_CONTINUOUSLY);
+        glView.setSurfaceRenderer(renderer);
+        return renderer;
+
+    }
+
+    private TangoConfig setupTangoConfig(Tango tango, boolean isAutoRecovery){
```

Comment:
Question: For simplicity, does this method even make sense to have?  And the parameter seems quite strange to me, considering this method is called once with a hardcoded value.